contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,604 @@
+
+1. The cluster that Bob uses has nodes that contain raw block devices that have not been formatted yet.
+2. The same addon DaemonSet can also discover block devices and creates corresponding PVs for them with the `block` field.  
+
+    ```yaml
+    kind: PersistentVolume
+    apiVersion: v1
+    metadata:
+      name: foo
+      labels:
+        kubernetes.io/hostname: node-1
+    spec:
+      capacity:
+        storage: 100Gi
+      local:
+        block:
+          device: /var/lib/kubelet/storage-raw-devices/foo
+      accessModes:
+        - ReadWriteOnce
+      persistentVolumeReclaimPolicy: Delete
+      storageClassName: local-fast
+    ```
+
+3. Bob creates a pod with a PVC that requests for block level access and similar to a Stateful Set scenario the scheduler will identify nodes that can satisfy the pods request.  The block devices will not be formatted to allow the application to handle the device using their own methods.
+
+    ```yaml
+    kind: PersistentVolumeClaim
+    apiVersion: v1
+    metadata:
+      name: myclaim
+    spec:
+      volumeType: block
+      storageClassName: local-fast
+      accessModes:
+        - ReadWriteOnce
+      resources:
+        requests:
+          storage: 80Gi
+    ```
+4. It is also possible for a PVC that requests `volumeType: file` to also use a block-based PV.  In this situation, the block device would get formatted with the filesystem type specified in the PV spec.  And when the PV gets destroyed, then the filesystem also gets destroyed to return back to the original block state.
+
+    ```yaml
+    kind: PersistentVolume
+    apiVersion: v1
+    metadata:
+      name: foo
+      labels:
+        kubernetes.io/hostname: node-1
+    spec:
+      capacity:
+        storage: 100Gi
+      local:
+        block:
+          path: /var/lib/kubelet/storage-raw-devices/foo
+          fsType: ext4
+      accessModes:
+        - ReadWriteOnce
+      persistentVolumeReclaimPolicy: Delete
+      storageClassName: local-fast
+    ```
+
+*The lifecycle of the block level PV will be similar to that of the scenarios explained earlier.* 
+
+# Open Questions & Discussion points 
+* Single vs split “limit” for storage across writable layer and logs
+    * Split allows for enforcement of hard quotas
+    * Single is a simpler UI
+* Local Persistent Volume bindings happening in the scheduler vs in PV controller
+    * Should the PV controller fold into the scheduler
+	* This will help spread PVs and pods across matching zones.
+* Repair/replace scenarios.
+    * What are the implications of removing a disk and replacing it with a new one? 
+    * We may not do anything in the system, but may need a special workflow
+* Volume-level replication use cases where there is no pod associated with a volume.  How could forgiveness/data gravity be handled there?
```

Comment:
Can you explain this case more thoroughly? Are you talking about the case where a node has a local PV, and some pod previously wrote some data into the PV, and the pod is now gone, but the claim is still in place (so the data hasn't been deleted), and you want to later schedule the replacement pod and make sure it ends up on this node? Can't that be handled just by having the pod reference the PVC? (I think you may have actually mentioned this earlier in the doc.)

How does forgiveness relate to this?